js/compiler/compiler.go

```diff
@@ -48,42 +48,58 @@ var (
 	once             sync.Once
 )
 
+// CompatibilityMode specifies the JS compatibility mode
+// nolint:lll
+//go:generate enumer -type=CompatibilityMode -transform=snake -trimprefix CompatibilityMode -output compatibility_mode_gen.go
+type CompatibilityMode uint
+
+// "es6": default, Goja+Babel+core.js
+// "es51": plain Goja
+const (
+	CompatibilityModeES6 CompatibilityMode = iota
+	CompatibilityModeES51
```

Comment:
It's in general good idea to start enums at `1` so that `0` is reserved for an invalid/not set value

Reply:
OK, I'll change it.

In this case I thought it was safer since "invalid/not set" doesn't make sense (we need a mode to run in), and any invalid values shouldn't make it here.

Reply:
Resolved by 1a75ee75.